zerver/lib/push_notifications.py

```diff
@@ -253,6 +263,11 @@ def parse_gcm_options(options: Dict[str, Any], data: Dict[str, Any]) -> str:
 
     return priority  # when this grows a second option, can make it a tuple
 
+def send_android_encrypted_push_notification(encrypted_data: EncryptedData,
+                                             options: Dict[str, Any], remote: bool=False) -> None:
+    for device, payload in encrypted_data:
+        send_android_push_notification([device], payload, options, remote)
+
```

Comment:
Why a loop here? We [discussed this in chat](https://chat.zulip.org/#narrow/stream/3-backend/topic/end-to-end.20encryption/near/931345) in a previous round. A single request with an `encrypted_payloads` field would make the code simpler. It would also make it easier to manage the case where a Zulip server for some reason ends up with lots of device records for a given user.

Reply:
I guess that approach would end up having us send a single large payload, rather than several small ones, in that buggy case?  Probably better since there's a high minimum latency, assuming we don't risk hitting a low limit for how large a payload can be.

Reply:
Ah I see, this isn't in the path leading to the bouncer and is only used when actually talking to FCM. I was thinking of the request the app server makes to the bouncer.

When actually talking to FCM or APNs, we have to make a separate request per distinct notification-message payload anyway, which means one per device when they're encrypted. So the loop here is appropriate -- it should just be inlined rather than being its own function, as mentioned in another comment.